runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -10135,9 +10133,11 @@ genHeapAlloc(TR::Node * node, TR::Instruction *& iCursor, bool isVariableLen, TR
             }
          }
 
-      if (!comp->getOption(TR_DisableDualTLH) && useDualTLH && node->canSkipZeroInitialization())
+      if (!comp->getOption(TR_DisableDualTLH) && node->canSkipZeroInitialization())
          {
-               iCursor = generateRXInstruction(cg, TR::InstOpCode::getCmpLogicalOpCode(), node, sizeReg,
+         cg->generateDebugCounter(TR::DebugCounter::debugCounterName(comp, "dualTLH/nonZeroAlloc/{%s}{%s}", comp->signature(),comp->getHotnessName(comp->getMethodHotness())),1,TR::DebugCounter::Undetermined);
```

Comment:
make sure there is no branching before this to other code path that allocate heap!

Reply:
move it to line 10153 to not count if calling helper